cmake/Modules/FindQhull.cmake

```diff
@@ -1,90 +1,194 @@
+#
+# ``QHULL::QHULL``
+#  Defined if the system has QHULL.
+#
+# Result Variables
+# ^^^^^^^^^^^^^^^^
+#
+# This module sets the following variables:
+#
+# ::
+#
+#   QHULL_FOUND               True in case QHULL is found, otherwise false
+#
+# Example usage
+# ^^^^^^^^^^^^^
+#
+# ::
+#
+#     find_package(QHULL REQUIRED)
+#
+#     add_executable(foo foo.cc)
+#     target_link_libraries(foo QHULL::QHULL)
 #
-# This sets the following variables:
-# QHULL_FOUND - True if QHULL was found.
-# QHULL_INCLUDE_DIRS - Directories containing the QHULL include files.
-# QHULL_LIBRARIES - Libraries needed to use QHULL.
-# QHULL_DEFINITIONS - Compiler flags for QHULL.
-# If QHULL_USE_STATIC is specified then look for static libraries ONLY else
-# look for shared ones
 
-if(QHULL_USE_STATIC)
-  set(QHULL_RELEASE_NAME qhullstatic_r)
-  set(QHULL_DEBUG_NAME qhullstatic_rd)
-else()
-  set(QHULL_RELEASE_NAME qhull_r qhull)
-  set(QHULL_DEBUG_NAME qhull_rd qhull_d)
+
+macro(ExtractProperties target)
+  get_target_property(QHULL_INTERFACE_INCLUDE_DIRECTORIES ${target} INTERFACE_INCLUDE_DIRECTORIES)
+  get_target_property(QHULL_INTERFACE_COMPILE_DEFINITIONS ${target} INTERFACE_COMPILE_DEFINITIONS)
+  get_target_property(QHULL_IMPORTED_CONFIGURATIONS ${target} IMPORTED_CONFIGURATIONS)
+  
+  foreach(config ${QHULL_IMPORTED_CONFIGURATIONS})
+    get_target_property(QHULL_IMPORTED_LOCATION_${config} ${target} IMPORTED_LOCATION_${config})
+    get_target_property(QHULL_IMPORTED_IMPLIB_${config} ${target} IMPORTED_IMPLIB_${config})
+  endforeach()
+endmacro()
+
+# Skip if QHULL::QHULL is already defined
+if(TARGET QHULL::QHULL)
+  return()
+endif()
+
+# Try to locate QHull using modern cmake config (available on latest Qhull version).
+find_package(Qhull CONFIG QUIET)
+
+if(Qhull_FOUND)
+  unset(Qhull_FOUND)
+  set(QHULL_FOUND ON)
+  set(HAVE_QHULL ON)
+  
+  message(STATUS "Found Qhull version ${Qhull_VERSION}")
+  
+  # Create interface library that effectively becomes an alias for the appropriate (static/dynamic) imported QHULL target
+  add_library(QHULL::QHULL INTERFACE IMPORTED)
+  
+  if(TARGET Qhull::qhull_r AND TARGET Qhull::qhullstatic_r)
+    if(PCL_QHULL_REQUIRED_TYPE MATCHES "DONTCARE")
+      if(PCL_SHARED_LIBS)
+        set_property(TARGET QHULL::QHULL APPEND PROPERTY INTERFACE_LINK_LIBRARIES Qhull::qhull_r)
+        set(QHULL_LIBRARY_TYPE SHARED)
+      else()
+        set_property(TARGET QHULL::QHULL APPEND PROPERTY INTERFACE_LINK_LIBRARIES Qhull::qhullstatic_r)
+        set(QHULL_LIBRARY_TYPE STATIC)
+      endif()
+    elseif(PCL_QHULL_REQUIRED_TYPE MATCHES "SHARED")
+      set_property(TARGET QHULL::QHULL APPEND PROPERTY INTERFACE_LINK_LIBRARIES Qhull::qhull_r)
+      set(QHULL_LIBRARY_TYPE SHARED)
+    else()
+      set_property(TARGET QHULL::QHULL APPEND PROPERTY INTERFACE_LINK_LIBRARIES Qhull::qhullstatic_r)
+      set(QHULL_LIBRARY_TYPE STATIC)
+    endif()
+  elseif(TARGET qhull::qhullstatic_r)
```

Comment:
Does capitalization matter? `qhull::` vs `Qhull::`?

Reply:
Yes. Thanks I'll fix it.